src/tsolvers/axdiffsolver/ArraySolver.cc

```diff
@@ -203,10 +203,10 @@ void ArraySolver::makeWeakRepresentative(NodeRef nodeRef) {
         // Invert primary edge
         getNode(parentRef).primaryEdge = nodeRef;
         getNode(parentRef).primaryStore = node.primaryStore;
-        node.primaryEdge = NodeRef_Undef;
         // Make representative for i-weak equivalence class
-        makeIndexedWeakRepresentative(nodeRef);
         // Information about primary store is needed in "makeIndexedWeakRepresentative"!
+        makeIndexedWeakRepresentative(nodeRef);
+        node.primaryEdge = NodeRef_Undef;
```

Comment:
I am not sure if it is safe to swap these operations. I'll have to think about it.

Reply:
I went through this a few times and it looked safe to me.  This is [proof by authority](http://jwilson.coe.uga.edu/EMT668/EMAT6680.F99/Challen/proof/proof.html) (which I unfortunately don't have :)

Reply:
OK, I am also convinced that this is safe, and even better then the original.